lib/templates/android_screen_base.tt

```diff
@@ -183,4 +183,25 @@ class AndroidScreenBase < Calabash::ABase
     # Clicking in the Done button
     touch "* id:'button1'"
   end
+  
+ 
+  
+  # @option [String] id The view's id (optional)
+  # @option [String] The complete view query
+  # @return [Array] return an array[String] associated with the views
+  def text_array_from(query, id = nil)
```

Comment:
If you have the complete query as a required parameter, why do you have the ID parameter? You just need to pass the ID inside the query.

We use a lot more IDs than queries. So to make easier to use, your first parameter should be an ID. If you have an ID, you make the query, if you don't have, you look for a complete query (optional parameter)

```
def text_array_from(id, query = nil)
```

And this is the standard of all other functions :)